src/zjs_sensor.h

```diff
@@ -1,9 +1,117 @@
-// Copyright (c) 2016, Intel Corporation.
+// Copyright (c) 2016-2017, Intel Corporation.
 
 #ifndef __zjs_sensor_h__
 #define __zjs_sensor_h__
 
+#include <sensor.h>
 #include "jerryscript.h"
+#include "zjs_callbacks.h"
+
+#define DEFAULT_SAMPLING_FREQUENCY      20
+#define SENSOR_MAX_CONTROLLER_NAME_LEN  16
+
+typedef enum sensor_state {
+    SENSOR_STATE_UNCONNECTED,
+    SENSOR_STATE_IDLE,
+    SENSOR_STATE_ACTIVATING,
+    SENSOR_STATE_ACTIVATED,
+    SENSOR_STATE_ERRORED,
+} sensor_state_t;
+
+typedef struct sensor_controller {
+    char name[SENSOR_MAX_CONTROLLER_NAME_LEN+1];
+    uint32_t pin;
+} sensor_controller_t;
+
+typedef struct sensor_handle {
+    sensor_state_t state;
+    enum sensor_channel channel;
+    sensor_controller_t *controller;
+    int frequency;
+    jerry_value_t sensor_obj;
+    zjs_callback_id onchange_cb_id;
+    zjs_callback_id onstart_cb_id;
+    zjs_callback_id onstop_cb_id;
+    struct sensor_handle *next;
+} sensor_handle_t;
+
+typedef struct sensor_instance {
+    void *handles;
+} sensor_instance_t;
```

Comment:
That's how I started with just returning the pointer, then it seems that I would need to store more stuff later, things like supported controllers, and maybe user_data, that's why I created the struct.